File: FullStack_Pagination/FootballApp/web/football-app-web/src/app/models/football-club-dto.ts
/**
 * Outbound representation of a football club.
 * Mirrors C# FootballClubDto exactly.
 */
export interface FootballClubDto {
    readonly id: number;
    readonly clubName: string;
    readonly country: string;
    readonly league: string;
    readonly foundedYear: number;
    readonly stadium: string;
    readonly titlesWon: number;
}

File: FootballApp/api/FootballApp.Api/Dtos/FootballClubDto.cs
namespace FootballApp.Api.Dtos;

/// <summary>
/// Outbound representation of a football club.
/// This is the public-facing shape returned by the API.
/// It is intentionally separate from the FootballClub entity so that
/// internal schema changes do not leak to the client.
/// </summary>
public sealed record FootballClubDto
(
    int Id,
    string ClubName,
    string Country,
    string League,
    int FoundedYear,
    string Stadium,
    int TitlesWon
);